transformer_engine/debug/features/log_fp8_tensor_stats.py

```diff
@@ -0,0 +1,155 @@
+# Copyright (c) 2022-2025, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# See LICENSE for license information.
+
+"""LogFp8TensorStats Feature support for nvidia-dlframework-inspect"""
+
+from typing import Dict, Union
+
+import torch
+
+import nvdlfw_inspect.api as debug_api
+from nvdlfw_inspect.debug_features.log_tensor_stats import LogTensorStats as BaseLogTensorStats
+from nvdlfw_inspect.registry import Registry, api_method
+
+from transformer_engine.debug.features.utils.stats_buffer import STATS_BUFFERS
+from transformer_engine.pytorch.tensor import QuantizedTensor
+from transformer_engine.pytorch.tensor.float8_tensor import Float8Tensor
+from transformer_engine.pytorch.tensor.mxfp8_tensor import MXFP8Tensor
+from transformer_engine.pytorch.tensor._internal.float8_tensor_base import Float8TensorBase
+from transformer_engine.pytorch.tensor._internal.mxfp8_tensor_base import MXFP8TensorBase
+from transformer_engine.debug.pytorch.debug_state import TEDebugState
+
+
+@Registry.register_feature(namespace="transformer_engine")
+class LogFp8TensorStats(BaseLogTensorStats):
+    """
+    This feature handles logging of FP8 tensor stats.
+
+
+    In a distributed setting, the auxiliary stats are computed on each rank and gathered after the `debug_api.step()` call. Do not forget to invoke `debug_api.step()` at every step to log stats!
+
+    `LogFp8TensorStats` supports micro-batching. If multiple forward/backward passes are invoked per `debug_api.step()`, then stats for all tensors except weights will be accumulated.
+
+    `LogFp8TensorStats` can induce significant overhead. To mitigate this issue, logging stats with `freq > 1` is recommended. If `LogFp8TensorStats` is not used in a given step, the overhead is smaller. Moreover, if no other feature is used for the layer, the TE layer will run as fast as it would without `debug_api` initialized.
```

Comment:
```suggestion
    In a distributed setting, the auxiliary stats are computed on each rank and gathered after
    the `debug_api.step()` call. Do not forget to invoke `debug_api.step()` at every step to log
    stats!

    `LogFp8TensorStats` supports micro-batching. If multiple forward/backward passes are invoked
    per `debug_api.step()`, then stats for all tensors except weights will be accumulated.

    `LogFp8TensorStats` can induce significant overhead. To mitigate this issue, logging stats
    with `freq > 1` is recommended. If `LogFp8TensorStats` is not used in a given step, the
    overhead is smaller. If no other feature is used for the layer, the TE layer will
    run as fast as it would without `debug_api` initialized.
```